src/software/new_geom/BUILD

```diff
@@ -4,30 +4,106 @@ cc_library(
     name = "new_geom",
     srcs = [
         "circle.cpp",
-        "line.cpp",
-        "point.cpp",
         "polynomial.cpp",
-        "vector.cpp",
     ],
     hdrs = [
-        "angle.h",
         "angular_velocity.h",
         "circle.h",
         "convex_shape.h",
-        "geom_constants.h",
-        "line.h",
-        "point.h",
         "polynomial.h",
         "ray.h",
         "shape.h",
-        "vector.h",
     ],
     deps = [
+        ":angle",
+        ":closest_point",
+        ":distance",
+        ":geom_constants",
+        ":line",
+        ":point",
+        ":vector",
+    ],
+)
+
+cc_library(
+    name = "angle",
+    hdrs = ["angle.h"],
+)
+
+cc_library(
+    name = "closest_point",
+    srcs = [
+        "util/closest_point.cpp",
+    ],
+    hdrs = [
+        "util/closest_point.h",
```

Comment:
In general, we try put libraries at the same level as the files they're composed of. Could we create a `BUILD` file in `util` and put this there (same goes for other libraries in `util` here).

Reply:
The libraries and tests in `util` are now in the new `BUILD` file in `util`. I also made other geometry classes (`Ray`, `Polynomial`, etc.) into their own libraries except shapes since more are being added right now.